lib/webmushra/pages/SpatialPage.js

```diff
@@ -0,0 +1,382 @@
+		}
+
+		var color = "#" + ((1 << 24) + respConfig.color).toString(16).substr(1);
+		button = $("<button id='" + respConfig.name + "' data-inline='true' style='background-color:#" + ((1 << 24) + respConfig.color).toString(16).substr(1) + "; background-image:radial-gradient(#ffffff 0%, " + color + " 100%)'>" + respConfig.label + "</button>");
+		button.on("click", (function() {
+			this.page.setBoxShadow(this.page.activeResponse, this.name, "#555555");
+			this.page.activeResponse = this.name;
+		}).bind({
+			page : this,
+			label : respConfig.label,
+			name : respConfig.name,
+			button : button
+		}));
+		$("#spatial_stimuli_" + stimulusIndex).append(button);
+	}
+	// scenes
+	var tableScenes = document.createElement("table");
+	tableScenes.setAttribute("border", "0");
+	tableScenes.setAttribute("align", "center");
+	tableScenes.setAttribute("style", "margin-top: 0em;");
+	var trScenes = document.createElement("tr");
+	for (var i = 0; i < this.pageConfig.views.length; ++i) {
+		var td = document.createElement("td");
+		td.appendChild(this.renderDivs[i]);
+		trScenes.appendChild(td);
+	}
+
+	tableScenes.appendChild(trScenes);
+	div.appendChild(tableScenes);
+
+	this.renderScene();
+
+	_id.append(div);
+};
+
+SpatialPage.prototype.setBoxShadow = function(_lastActiveButton, _currentActiveButton, color) {
+	if (_lastActiveButton != null) {
+		document.getElementById(_lastActiveButton).style.boxShadow = "0 0 0 white , 0 0 0 white";
+
+	}
+	document.getElementById(_currentActiveButton).style.boxShadow = "0 0 30px " + color + " , 0 0 30px " + color;
+};
+
+SpatialPage.prototype.renderScene = function() {
+	for (var i = 0; i < this.pageConfig.views.length; ++i) {
+		this.scenes[i].render();
+	}
+};
+
+SpatialPage.prototype.store = function() {
+  var trial_localization = this.session.getTrial("localization", this.pageConfig.id);
+ 
+  var trial_asw = this.session.getTrial("asw", this.pageConfig.id);
+ 
+  var trial_lev = this.session.getTrial("lev", this.pageConfig.id);
+ 
+  var trial_hwd = this.session.getTrial("hwd", this.pageConfig.id);
+  
+  
+	for (key in this.responses) { 
+		if(this.responses[key].objects[0] instanceof (ASWObject) && key.height == 0){
+			this.responses[key].objects[0].storeValues(trial_asw, this.session, this.pageConfig.id);
+		}
+		if(this.responses[key].objects[0] instanceof (LocalizationObject)){
+			this.responses[key].objects[0].storeValues(trial_localization, this.session, this.pageConfig.id);
+		}
+		if(this.responses[key].objects[0] instanceof (LEVObject)){
+			this.responses[key].objects[0].storeValues(trial_lev, this.session, this.pageConfig.id);
+		}
+		if(this.responses[key].objects[0] instanceof (ASWObject)){
+			this.responses[key].objects[0].storeValues(trial_hwd, this.session, this.pageConfig.id);
+		}
+		
+	}
+};
+
+SpatialPage.prototype.load = function() {
+	this.startTimeOnPage = new Date();
+	this.frameUpdateInterval = setInterval((function() {
+		this.renderScene();
+	}).bind(this), this.framerate);

```

Comment:
Ok, I added this to my TODO list and will implement it in another commit (This new feature has some other similar things that I do not like)